File: presentation/src/main/java/com/mbmc/template/Application.java
package com.mbmc.template;

import android.app.Activity;
import com.mbmc.template.di.ApplicationComponent;
import com.mbmc.template.di.DaggerApplicationComponent;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;

import javax.inject.Inject;

public class Application extends android.app.Application implements HasActivityInjector {

    @Inject DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        getComponent().inject(this);
    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidInjector;
    }

    protected ApplicationComponent getComponent() {
        return DaggerApplicationComponent.create();
    }

}

File: presentation/src/main/kotlin/com/mbmc/template/Application.kt
package com.mbmc.template

import android.app.Activity
import com.mbmc.template.di.component.ApplicationComponent
import com.mbmc.template.di.component.DaggerApplicationComponent
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector

import javax.inject.Inject

open class Application : android.app.Application(), HasActivityInjector {
    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

    override fun onCreate() {
        super.onCreate()
        getComponent().inject(this)
    }

    override fun activityInjector(): AndroidInjector<Activity> = dispatchingAndroidInjector

    open fun getComponent(): ApplicationComponent = DaggerApplicationComponent.create()
}
